### DISCUSSION TITLE
Login Error: Undefined class constant 'ALNUM2'

### DISCUSSION THREAD

**Discussion by TonyM15** [YetiForceCompany/YetiForceCRM#16626] [General]
Suddenly system start showing below error while login:
2022-11-03 18:38:40.0264 [error][] - Undefined class constant 'ALNUM2'
Error: Undefined class constant 'ALNUM2' in /var/www/html/modules/Users/actions/Login.php:178

Commenting above line fixed it, but left me confused as cannot detect what may cause it. 
Does anybody faced the same?

**Comment by rskrzypczak** [comment]
You've probably changed files manually.
Perhaps the login page or app\Purifier.php.
It could also be caused by an incorrect system update and only some of the files were updated.
However, you have a file mismatch because in some files you are referring to elements that do not exist.
You are probably the only one who will be able to pinpoint the cause of this. You need to track the history of modifications in your instance.
